src/app/components/user-profile/user-profile.component.ts

```diff
@@ -10,46 +12,82 @@ import {MyOption} from '../../model/MyOption';
-    this.getFirstName().valueChanges
-      .subscribe(firstName => console.log('Full name value changed: ', firstName));
+    const getCanada = mapCountryTranslate.get(CountriesEnum.CANADA);
+    const getUS = mapCountryTranslate.get(CountriesEnum.US);
+    const getQC = mapProvinceTranslate.get(ProvincesEnum.QUEBEC);
+    const getBC = mapProvinceTranslate.get(ProvincesEnum.BC);
+    const getON = mapProvinceTranslate.get(ProvincesEnum.ONTARIO);
+    const getTX = mapProvinceTranslate.get(ProvincesEnum.TEXAS);
+    const getCALI = mapProvinceTranslate.get(ProvincesEnum.CALIFORNIA);
+    const getFL = mapProvinceTranslate.get(ProvincesEnum.FLORIDA);
+
+
+    this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
+      this.translateService.get([getCanada, getUS])
+        .subscribe((countriesTranslation) => {
+          this.countries = [{label: countriesTranslation[getCanada], value: CountriesEnum.CANADA},
+            {label: countriesTranslation[getUS], value: CountriesEnum.US}];
+        });
+    });
+
 
+    this.userProfileForm = this.createFormGroup();
     this.userProfileForm.get('dropdownGroup.country').valueChanges
       .subscribe(country => {
         if (country) {
-          this.userProfileForm.get('dropdownGroup.province').patchValue('');
-          console.log('CountriesEnum', CountriesEnum);
-          console.log('country', country);
-          this.provinces = mapProvinces.get(country)
-            .map(province => ({value: province, label: province}));
-          console.log('this.provinces', this.provinces);
+          this.translateService.get(mapProvinces.get(country))
+            .subscribe((provinceTranslation) => {
+
+              // whole point of mapping was not to do this
+              if (country === 'CANADA') {
+                this.provinces = [
+                  {label: provinceTranslation[getQC], value: ProvincesEnum.QUEBEC},
+                  {label: provinceTranslation[getON], value: ProvincesEnum.ONTARIO},
+                  {label: provinceTranslation[getBC], value: ProvincesEnum.BC}
+                ];
+              } else {
+                this.provinces = [
+                  {label: provinceTranslation[getCALI], value: ProvincesEnum.CALIFORNIA},
+                  {label: provinceTranslation[getTX], value: ProvincesEnum.TEXAS},
+                  {label: provinceTranslation[getFL], value: ProvincesEnum.FLORIDA}
+                ];
+              }
+            });
         } else {
-          console.log('No country');
+          console.warn('No country');
         }
       });
 
-    this.userProfileForm.get('dropdownGroup.province').valueChanges
-      .subscribe(province => console.log('Province value changed: ', province));
+    this.userProfileForm.valueChanges
+      .subscribe(formData => {
+        this.formValues.name = formData.firstName + ' ' + formData.lastName;
+        this.formValues.country = formData.dropdownGroup.country;
```

Comment:
As per requirement ` The country name selected .... but in english, whatever is the current lang` ...
You are giving the value here not the name of the country